aviary/interface/methods_for_level2.py

```diff
@@ -1237,478 +1244,513 @@ def run_aviary_problem(
 
-        if phase_info is None:
-            # model.phase_info only contains mission information
-            phase_info = self.model.phase_info
-            phase_info['pre_mission'] = self.model.pre_mission_info
-            phase_info['post_mission'] = self.model.post_mission_info
-        if mission_range is None:
-            # mission range is sliced from a column vector numpy array, i.e. it is a len
-            # 1 numpy array
-            # fixes issue wherein an alternate mission with no inputs "alternate_mission()" attempts to run the range
-            # within the CSV file instead of in the phase_info.
-            if mass_method == LegacyCode.GASP:
-                mission_range = self.get_val(Mission.Design.RANGE)[0]
-            elif mass_method == LegacyCode.FLOPS:
-                try:
-                    mission_range = self.model.post_mission_info['target_range'][0]
-                except:
-                    mission_range = self.get_val(Mission.Design.RANGE)[0]
-
-        # gross mass is sliced from a column vector numpy array, i.e. it is a len 1 numpy
-        # array
-        mission_mass = self.get_val(Mission.Design.GROSS_MASS)[0]
-        optimizer = self.driver.options['optimizer']
-
-        prob_alternate = _load_off_design(
-            json_filename,
-            ProblemType.ALTERNATE,
-            equations_of_motion,
-            mass_method,
-            phase_info,
-            num_first,
-            num_business,
-            num_tourist,
-            num_pax,
-            wing_cargo,
-            misc_cargo,
-            cargo_mass,
-            mission_range,
-            mission_mass,
-        )
+            off_design_prob.aviary_inputs.set_val(
+                Mission.Summary.GROSS_MASS, mission_gross_mass, units='lbm'
+            )
 
-        # TODO: All these methods will need to be updated
-        prob_alternate.check_and_preprocess_inputs()
-        prob_alternate.build_model()
-        prob_alternate.add_driver(optimizer, verbosity=verbosity)
-        prob_alternate.options = self.options
-        prob_alternate.driver.options = self.driver.options
-        prob_alternate.driver.opt_settings = self.driver.opt_settings
-        prob_alternate.add_design_variables()
-        prob_alternate.add_objective()
-        prob_alternate.setup()
-        if run_mission:
-            prob_alternate.run_aviary_problem()
-        return prob_alternate
-
-    def fallout_mission(
-        self,
-        run_mission=True,
-        json_filename='sizing_problem.json',
-        num_first=None,
-        num_business=None,
-        num_tourist=None,
-        num_pax=None,
-        wing_cargo=None,
-        misc_cargo=None,
-        cargo_mass=None,
-        mission_mass=None,
-        phase_info=None,
-        verbosity=None,
-    ):
+        off_design_prob.check_and_preprocess_inputs(verbosity=verbosity)
+        off_design_prob.add_pre_mission_systems(verbosity=verbosity)
```

Comment:
Remember you can use `off_design_prob.build_model()` to execute add_pre, add_phase, add_post, and link_phases!